eyeshade/controllers/rates.js

```diff
@@ -0,0 +1,43 @@
+const Joi = require('joi')
+const boom = require('boom')
+const underscore = require('underscore')
+
+const utils = require('bat-utils')
+const braveHapi = utils.extras.hapi
+
+const v1 = {}
+
+/*
+   GET /v1/rates
+ */
+
+v1.read = { handler: (runtime) => {
+  return async (request, reply) => {
+    const result = { altrates: runtime.currency.altrates, fxrates: runtime.currency.fxrates, rates: runtime.currency.rates }
+    let validity
+
+    underscore.keys(result).forEach((key) => { if (underscore.keys(result[key]).length === 0) delete result[key] })
+    validity = Joi.validate(result, runtime.currency.schemas.rates)
```

Comment:
is there a reason to explicitly validate here vs using the response validation below?

Reply:
another good catch! there's a hook that i can use.